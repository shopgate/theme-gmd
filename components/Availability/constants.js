/**
 * Copyright (c) 2017, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

/**
 * The availability states of a product.
 */
export const AVAILABILITY_STATE_OK = 'ok';
export const AVAILABILITY_STATE_WARNING = 'warning';
export const AVAILABILITY_STATE_ALERT = 'alert';
