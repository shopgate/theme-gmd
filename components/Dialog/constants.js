/**
 * Copyright (c) 2017-present, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

export const DIALOG_TEXT_MESSAGE = 'DIALOG_TEXT_MESSAGE';
export const MODAL_VARIANT_SELECT = 'MODAL_VARIANT_SELECT';
