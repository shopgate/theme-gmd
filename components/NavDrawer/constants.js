/**
 * Copyright (c) 2017-present, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

export const QUICKLINKS_MENU = 'quicklinks';
