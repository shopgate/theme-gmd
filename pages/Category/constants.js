/**
 * Copyright (c) 2017-present, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

export const GRID_VIEW = 'grid';
export const LIST_VIEW = 'list';

export const SET_CATEGORY_VIEW_MODE = 'SET_CATEGORY_VIEW_MODE';
