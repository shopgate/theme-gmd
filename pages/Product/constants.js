/**
 * Copyright (c) 2017-present, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

/**
 * Action Types
 */
export const OPEN_PRODUCT_GALLERY = 'OPEN_PRODUCT_GALLERY';
export const REQUEST_PRODUCT_DATA = 'REQUEST_PRODUCT_DATA';

/**
 * Max number of reviews shown
 * @type {number}
 */
export const REVIEW_PREVIEW_COUNT = 2;
