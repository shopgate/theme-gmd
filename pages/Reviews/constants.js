/**
 * Copyright (c) 2017-present, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

/**
 * Number of reviews items that should be loaded per bundle (page or load more).
 * @type {number}
 */
export const REVIEW_ITEMS_PER_PAGE = 10;
