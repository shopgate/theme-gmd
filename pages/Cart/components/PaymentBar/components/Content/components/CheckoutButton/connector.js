/**
 * Copyright (c) 2017-present, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import { connect } from 'react-redux';
import pushHistory from '@shopgate/pwa-common/actions/history/pushHistory';
import { isViewLoading } from '@shopgate/pwa-common/selectors/view';
import { CART_PATH } from '@shopgate/pwa-common-commerce/cart/constants';
import { getOrderableStatus } from '@shopgate/pwa-common-commerce/cart/selectors';

/**
 * Maps the contents of the state to the component props.
 * @param {Object} state The current application state.
 * @return {Object} The extended component props.
 */
const mapStateToProps = state => ({
  isActive: getOrderableStatus(state) && !isViewLoading(state, CART_PATH),
});

/**
 * Connects the dispatch function to a callable function in the props.
 * @param {Function} dispatch The redux dispatch function.
 * @return {Object} The extended component props.
 */
const mapDispatchToProps = dispatch => ({
  pushHistory: url => dispatch(pushHistory(url)),
});

export default connect(mapStateToProps, mapDispatchToProps);
