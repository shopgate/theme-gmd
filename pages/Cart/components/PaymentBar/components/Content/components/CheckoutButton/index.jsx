/**
 * Copyright (c) 2017-present, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { CHECKOUT_PATH } from '@shopgate/pwa-common/constants/RoutePaths';
import I18n from '@shopgate/pwa-common/components/I18n';
import RippleButton from 'Components/RippleButton';
import connect from './connector';

/**
 * The checkout button component.
 */
class CheckoutButton extends Component {
  static propTypes = {
    pushHistory: PropTypes.func.isRequired,
    isActive: PropTypes.bool,
  };

  static defaultProps = {
    isActive: true,
  };

  static clickDelay = 300;

  /**
   * Go to the checkout
   */
  handleClick = () => {
    setTimeout(() => this.props.pushHistory(CHECKOUT_PATH), CheckoutButton.clickDelay);
  };

  /**
   * The render function.
   * @returns {JSX}
   */
  render() {
    return (
      <RippleButton
        onClick={this.handleClick}
        disabled={!this.props.isActive}
        flat={false}
        type="secondary"
      >
        <I18n.Text string="cart.checkout" />
      </RippleButton>
    );
  }
}

export default connect(CheckoutButton);
