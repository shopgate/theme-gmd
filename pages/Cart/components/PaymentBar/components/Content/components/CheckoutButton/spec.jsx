/**
 * Copyright (c) 2017-present, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React from 'react';
import { shallow } from 'enzyme';
import { CHECKOUT_PATH } from '@shopgate/pwa-common/constants/RoutePaths';
import RippleButton from 'Components/RippleButton';
import CheckoutButton from './index';

// Mock the history connector
jest.mock('./connector', () => (obj) => {
  const newObj = obj;

  newObj.defaultProps = {
    ...newObj.defaultProps,
    pushHistory: () => {},
    isActive: false,
  };

  return newObj;
});

describe('<CheckoutButton /', () => {
  jest.useFakeTimers();

  let wrapper;
  let childButton;

  describe('Checkout is possible', () => {
    beforeEach(() => {
      wrapper = shallow(
        <CheckoutButton />
      );

      childButton = wrapper.find(RippleButton);
    });

    it('should render without any props', () => {
      expect(wrapper).toMatchSnapshot();
      expect(childButton.props().disabled).toBe(true);
    });

    it('should trigger a pushHistory on click', () => {
      const pushHistorySpy = jest.fn();

      wrapper.setProps({
        pushHistory: pushHistorySpy,
      });

      childButton.simulate('click');

      jest.runOnlyPendingTimers();

      // Check if the command was called
      expect(pushHistorySpy).toHaveBeenCalledTimes(1);
      expect(pushHistorySpy).toBeCalledWith(CHECKOUT_PATH);

      // Check if the execution of  click handler was delayed
      expect(setTimeout.mock.calls.length).toBe(1);
      expect(setTimeout.mock.calls[0][1]).toBe(CheckoutButton.clickDelay);
    });
  });

  describe('Checkout is not possible', () => {
    beforeEach(() => {
      wrapper = shallow(
        <CheckoutButton checkoutPossible={false} />
      );

      childButton = wrapper.find(RippleButton);
    });

    it('should not trigger the pushHistory on click', () => {
      const pushHistorySpy = jest.fn();

      wrapper.setProps({
        pushHistory: pushHistorySpy,
      });

      childButton.simulate('click');

      expect(pushHistorySpy).toHaveBeenCalledTimes(0);
    });
  });
});
