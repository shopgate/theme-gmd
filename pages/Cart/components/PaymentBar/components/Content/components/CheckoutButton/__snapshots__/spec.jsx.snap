// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CheckoutButton /> Checkout is possible should render without any props 1`] = `
<Link
  className=""
  disabled={true}
  href="/checkout"
>
  <RippleButton
    className=""
    disabled={true}
    flat={false}
    onClick={null}
    rippleClassName=""
    rippleSize={null}
    testId="Button"
    type="secondary"
    wrapContent={true}
  >
    <Translate
      className=""
      params={Object {}}
      string="cart.checkout"
    />
  </RippleButton>
</Link>
`;
